### PR TITLE
Fix broken relative links

### PR DESCRIPTION
**Pull request by winkerVSbecks**
### What's the issue?

We have relative links in documentation pages. Eg: the first link on this page https://storybook.js.org/docs/react/writing-docs/doc-blocks/ is `../addons/writing-addons`

When the site loads up, you can click and navigate correctly. However, in the raw html `<a href="../addons/writing-addons">addons</a>` navigates to https://storybook.js.org/docs/react/writing-docs/addons/writing-addons which doesn't exist. Which is what bots crawl and we have a bunch of SEO errors.

For some reason the directory structure of the static build is different to the way relative links are defined in Gatsby.

### What I did?
Added a component to the MDX renderer to turn:

* `../something` type links into `/docs/framework/something`
* `./something` type links into `something`
* `../../something` get converted into github links so, remain as is

### CODE REVIEW

**Comment by domyen** [comment]
**Bug/edge case:** 
[This page](https://github.com/storybookjs/storybook/blob/next/docs/writing-docs/introduction.md) has relative linkes.
> If you're including Storybook in your project for the **first time**, we provide you with **DocsPage**...

The first, in bold, works fine in both GitHub and our docs. The second, DocsPage, works in GitHub but doesn't work in the [deploy preview](https://deploy-preview-232--storybook-frontpage.netlify.app/docs/react/writing-docs/introduction). It appears like the framework `react` is getting appended to the url.

**Comment by winkerVSbecks** [comment]
@domyen yea, I noticed that too. I just pushed a fix.

**Comment by domyen** [comment]
I'm seeing the same behavior for some reason. `react` get's appended to the path like so `https://deploy-preview-232--storybook-frontpage.netlify.app/docs/react/writing-docs/docs-page/**react**`

**Comment by winkerVSbecks** [comment]
I see, I was confusing this with another issue I found.

File: gatsby-config.js [hunk 1]
```diff
@@ -118,6 +118,7 @@ module.exports = {
               isIconAfterHeader: false,
             },
           },
+          'gatsby-remark-check-links',
```
Comment (kylesuss) [inline gatsby-config.js:121]: Nice

**Review by kylesuss — COMMENTED** [review]
Using Dom's example:

https://deploy-preview-232--storybook-frontpage.netlify.app/docs/react/writing-docs/introduction

Screen Shot 2021-02-26 at 3 55 06 PM

Is that expected?

**Comment by winkerVSbecks** [comment]
@kylesuss no, I stuck with that one. Going to take another pass at it.

**Review by kylesuss — APPROVED** [review]
Links LGTM on latest update 👍

**Comment by domyen** [comment]
lgtm!

**Review by domyen — APPROVED** [review]

**Pull request merged**